src/v/ssx/tests/async_transforms.cc

```diff
@@ -54,9 +54,7 @@ SEASTAR_THREAD_TEST_CASE(async_transform_move_test) {
     std::vector<ss::sstring> expected = ssx::async_transform(
                                           input_copy.begin(),
                                           input_copy.end(),
-                                          [](ss::sstring s) {
-                                              return std::move(s);
-                                          })
+                                          [](ss::sstring s) { return s; })
```

Comment:
I didn't think this would move/elide the copy, but there's a test that it works with noncopyable types, so I guess it's good.

Reply:
it won't elide the copy because s is a parameter, but iiuc s will be treated as an rvalue in this case for the purposes of selecting the constructor for the 'copy':

> 12.8/32:
> When the criteria for elision of a copy operation are met or would be met save for the fact that the source object is a function parameter, and the object to be copied is designated by an lvalue, overload resolution to select the constructor for the copy is ﬁrst performed as if the object were designated by an rvalue.

at least that was my interpretation based on this SO answer https://stackoverflow.com/a/14856553